internal/cloud/common.go

```diff
@@ -0,0 +1,51 @@
+package cloud
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Common struct {
+	ClusterName       string     `env:"CLUSTER_NAME" validate:"required"`
+	ArtifactBucketURL *BucketURL `env:"ARTIFACT_BUCKET_URL,noinit" validate:"required"`
+	RegistryURL       string     `env:"REGISTRY_URL" validate:"required"`
```

Comment:
This gave an idea here: https://github.com/substratusai/substratus/issues/112

Reply:
Agreed, as the code is written now, `REGISTRY_URL` should take precedence over the auto-configured GCR URL.